Packages/com.unity.inputsystem/InputSystem/Plugins/XInput/XInputControllerWindows.cs

```diff
@@ -93,6 +93,7 @@ namespace UnityEngine.InputSystem.XInput
     /// An <see cref="XInputController"/> compatible game controller connected to a Windows desktop machine.
     /// </summary>
     [InputControlLayout(stateType = typeof(XInputControllerWindowsState), hideInUI = true)]
+    [Scripting.Preserve]
```

Comment:
Makes me wonder, do you think it would make sense to derive `InputControlLayoutAttribute` from `PreserveAttribute`? Seems a little hackish and not sure it'd even work. If we did it, probably would make sense for `InputControlAttribute`, too. For the latter we'd have to find another solution to it being co-opted as a `PropertyAttribute`.

On the upside, seems like this would implicitly solve the problem for users, too.

Reply:
That was my first thought as well. But it is not supported.

Reply:
Ugh, sad :(